cc-backend/controllers/addressController.ts

```
@@ -0,0 +1,161 @@
+    
+    try {
+        const userId = req.user.id;
+        
+        const addresses = await initGetAddress(userId);
+        
+        return res.status(STATUS_CODE.HTTP_OK).json({
+            message: 'Addresses retrieved successfully',
+            addresses: addresses,
+            count: addresses.length
+        });
+
+    } catch (error) {
+        logError(`Error getting addresses: ${error.message}`);
+        return res.status(STATUS_CODE.HTTP_INTERNAL_SERVER_ERROR).json({
+            error: 'Internal server error'
+        });
+    }
+};
+
+// Update address
+export const handleUpdateAddress = async (req: Request, res: Response) => {
+    logInfo("entered handleUpdateAddress");
+    
+    try {
+        const addressId = req.params.id;
+        if (!addressId) {
+            return res.status(STATUS_CODE.HTTP_BAD_REQUEST).json({
+                error: 'Address ID is required'
+            });
+        }
+
+        const validation = UpdateAddressSchema.safeParse(req.body);
+        if (!validation.success) {
+            const message = validation.error.issues.map(issue => issue.message).join(', ');
+            return res.status(STATUS_CODE.HTTP_BAD_REQUEST).json({
+                error: `Validation failed: ${message}`
+            });
+        }
+
+        const userId = req.user.id;
+
+        const updatedAddress = await initUpdateAddress(addressId, validation.data, userId);
+        
+        if (!updatedAddress) {
+            return res.status(STATUS_CODE.HTTP_NOT_FOUND).json({
+                error: 'Address not found or access denied'
+            });
+        }
+
+        return res.status(STATUS_CODE.HTTP_OK).json({
+            message: 'Address updated successfully',
+            address: updatedAddress
+        });
+
+    } catch (error) {
+        logError(`Error updating address: ${error.message}`);
+        return res.status(STATUS_CODE.HTTP_INTERNAL_SERVER_ERROR).json({
+            error: 'Internal server error'
+        });
```

I see that for all of your 500 responses in this file, you are returning the string "Internal server error" as the value to error. to make the error responses more granular, lets return the actual error message so that the response is more descriptive of what went wrong